lib/archethic/transaction_chain/transaction_input.ex

```diff
@@ -219,8 +239,24 @@ defmodule Archethic.TransactionChain.TransactionInput do
       from: from,
       type: :call,
       spent: spent?,
-      reward: false,
       timestamp: timestamp
     }
   end
+
+  def to_map(%__MODULE__{
+        amount: _,
+        from: from,
+        spent?: spent?,
+        type: :state,
+        timestamp: timestamp,
+        encoded_payload: encoded_payload
+      }) do
+    %{
+      from: from,
+      type: :state,
+      spent: spent?,
+      timestamp: timestamp,
+      encoded_payload: encoded_payload
```

Comment:
I think this should have the same behavior of unspent output, returning the deserialized state